File: examples/java/sdk-reference/error-handling/validation-error.java
import java.time.Duration;
import java.util.Map;
import software.amazon.lambda.durable.DurableContext;
import software.amazon.lambda.durable.DurableHandler;

public class ValidationErrorExample extends DurableHandler<Map<String, Object>, Map<String, Object>> {

    @Override
    public Map<String, Object> handleRequest(Map<String, Object> event, DurableContext context) {
        // The SDK throws IllegalArgumentException for invalid configuration values.
        // For example, passing a duration shorter than 1 second to context.wait():
        try {
            context.wait("short-wait", Duration.ofMillis(500)); // invalid: less than 1 second
        } catch (IllegalArgumentException e) {
            context.getLogger().error("Invalid SDK configuration: " + e.getMessage());
            return Map.of("error", "InvalidConfiguration", "message", e.getMessage());
        }
        return Map.of("status", "ok");
    }
}

File: examples/csharp/sdk-reference/error-handling/validation-error.cs
using Amazon.Lambda.Core;
using Amazon.Lambda.DurableExecution;
using Microsoft.Extensions.Logging;

public class ValidationErrorExample
{
    public Task<DurableExecutionInvocationOutput> Handler(
        DurableExecutionInvocationInput input, ILambdaContext context)
        => DurableFunction.WrapAsync<object, ValidationResult>(Workflow, input, context);

    private async Task<ValidationResult> Workflow(object input, IDurableContext ctx)
    {
        // The SDK throws ArgumentOutOfRangeException for invalid configuration
        // values. For example, passing a duration shorter than 1 second to
        // WaitAsync:
        try
        {
            await ctx.WaitAsync(TimeSpan.FromMilliseconds(500), name: "short-wait"); // invalid: less than 1 second
        }
        catch (ArgumentOutOfRangeException e)
        {
            ctx.Logger.LogError("Invalid SDK configuration: {Message}", e.Message);
            return new ValidationResult("InvalidConfiguration", e.Message);
        }
        return new ValidationResult("ok", null);
    }
}

public record ValidationResult(string Status, string? Message);
